# 실습1

In [1]:
import pandas as pd

In [2]:
df = pd.read_excel('iqtest.xlsx')

In [3]:
df.head()

Visual  Cubes  Lozenges  Comprehension  Completion  Word  Addition  \
0      23     19         4             10          17    10        69   
1      33     22        17              8          17    10        65   
2      34     24        22             11          19    19        50   
3      29     23         9              9          19    11       114   
4      16     25        10              8          25    24       112   

   Counting  Curve  
0        82    156  
1        98    195  
2        86    228  
3       103    144  
4       122    160

In [4]:
from statsmodels.formula.api import ols

In [5]:
ols('Comprehension ~ Word', data = df).fit().summary()
# Comprehensiom을 종속변수(y). Word를 독립변수(x)로 회귀분석을 한다

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:          Comprehension   R-squared:                       0.510
Model:                            OLS   Adj. R-squared:                  0.507
Method:                 Least Squares   F-statistic:                     149.1
Date:                Tue, 06 Oct 2020   Prob (F-statistic):           6.11e-24
Time:                        14:54:44   Log-Likelihood:                -329.85
No. Observations:                 145   AIC:                             663.7
Df Residuals:                     143   BIC:                             669.7
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      4.7073      0.472      9.964      0.000       3.773       5.641
Word           0.3034      0.025     12.211      0.000       0.254       0.353
==============================================================================
Omnibus:                        1.390   Durbin-Watson:                   1.998
Prob(Omnibus):                  0.499   Jarque-Bera (JB):                0.996
Skew:                          -0.022   Prob(JB):                        0.608
Kurtosis:                       3.404   Cond. No.                         45.7
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

- 기울기와 절편이 중요
- 결과를 해석할 줄 알아야한다: 단어를 많이 알면 독해점수가 높아진다
- 세개의 표 
    - 가장 위에 있는 표는 분석 전체에 관련된 것
    - 두번째 표는 계수와 관련된 정보
    - 마지막 표는 잔차와 관련된 정보
- R-squared: Word가 Comprehension의 분산에서 51%를 설명한다

# 단순회귀분석 실습

In [6]:
df = pd.read_excel('speed.xlsx')

In [7]:
df.head()

speed  dist
0      4     2
1      4    10
2      7     4
3      7    22
4      8    16

In [8]:
ols('dist ~ speed', data = df).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   dist   R-squared:                       0.651
Model:                            OLS   Adj. R-squared:                  0.644
Method:                 Least Squares   F-statistic:                     89.57
Date:                Tue, 06 Oct 2020   Prob (F-statistic):           1.49e-12
Time:                        14:54:45   Log-Likelihood:                -206.58
No. Observations:                  50   AIC:                             417.2
Df Residuals:                      48   BIC:                             421.0
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept    -17.5791      6.758     -2.601      0.012     -31.168      -3.990
speed          3.9324      0.416      9.464      0.000       3.097       4.768
==============================================================================
Omnibus:                        8.975   Durbin-Watson:                   1.676
Prob(Omnibus):                  0.011   Jarque-Bera (JB):                8.189
Skew:                           0.885   Prob(JB):                       0.0167
Kurtosis:                       3.893   Cond. No.                         50.7
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

**해석하기**

- 제동거리 데이터를 다운받아 speed를 독립변수로 dist를 종속변수로 회귀분석을 해보십시오. R제곱은 얼마입니까?
    - 0.651

- 위의 회귀분석에서 절편은 얼마입니까?
    - -17.5791
    - 속도가 0일때를 가정

- 위의 회귀분석에서 speed의 기울기는 얼마입니까?
    - 3.9324
  
- 신뢰구간
    - 3.097~4.768


In [9]:
df3 = pd.read_excel('safety.xlsx')

In [10]:
ols('risk ~ fuel', data = df3).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   risk   R-squared:                       0.027
Model:                            OLS   Adj. R-squared:                  0.021
Method:                 Least Squares   F-statistic:                     4.313
Date:                Tue, 06 Oct 2020   Prob (F-statistic):             0.0395
Time:                        14:54:45   Log-Likelihood:                -251.03
No. Observations:                 159   AIC:                             506.1
Df Residuals:                     157   BIC:                             512.2
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
===============================================================================
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
Intercept       0.1333      0.305      0.437      0.662      -0.469       0.736
fuel[T.gas]     0.6653      0.320      2.077      0.039       0.033       1.298
==============================================================================
Omnibus:                        2.051   Durbin-Watson:                   1.005
Prob(Omnibus):                  0.359   Jarque-Bera (JB):                1.659
Skew:                           0.088   Prob(JB):                        0.436
Kurtosis:                       2.532   Cond. No.                         6.36
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

- T.gas = gas - disel
- 0.6653 -> 휘발유(가솔린) 차가 더 위험하군.
- 진짜일까? 신뢰구간을 확인
    - 모집단에서는 0.033~1.208만큼 위험할 수 있으나 어쨌든 부호가 +니까 위험한 것은 맞음
- 가솔린의 위험도 = 0.6653 + 0.1333

In [11]:
ols('risk ~ wheels', data = df3).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   risk   R-squared:                       0.035
Model:                            OLS   Adj. R-squared:                  0.023
Method:                 Least Squares   F-statistic:                     2.842
Date:                Tue, 06 Oct 2020   Prob (F-statistic):             0.0613
Time:                        14:54:45   Log-Likelihood:                -250.34
No. Observations:                 159   AIC:                             506.7
Df Residuals:                     156   BIC:                             515.9
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         0.5000      0.417      1.199      0.232      -0.324       1.324
wheels[T.fwd]     0.3952      0.433      0.914      0.362      -0.459       1.250
wheels[T.rwd]    -0.0870      0.452     -0.192      0.848      -0.979       0.805
==============================================================================
Omnibus:                        4.022   Durbin-Watson:                   1.030
Prob(Omnibus):                  0.134   Jarque-Bera (JB):                3.757
Skew:                           0.311   Prob(JB):                        0.153
Kurtosis:                       2.577   Cond. No.                         9.76
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

# 다중회귀분석

연습

In [12]:
ols('risk ~ length + height + wheels', data = df3).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   risk   R-squared:                       0.244
Model:                            OLS   Adj. R-squared:                  0.224
Method:                 Least Squares   F-statistic:                     12.43
Date:                Tue, 06 Oct 2020   Prob (F-statistic):           8.71e-09
Time:                        14:54:45   Log-Likelihood:                -230.94
No. Observations:                 159   AIC:                             471.9
Df Residuals:                     154   BIC:                             487.2
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept        14.5514      2.194      6.632      0.000      10.217      18.886
wheels[T.fwd]    -0.1356      0.396     -0.343      0.732      -0.917       0.646
wheels[T.rwd]    -0.3488      0.437     -0.798      0.426      -1.212       0.514
length           -0.0075      0.010     -0.729      0.467      -0.028       0.013
height           -0.2286      0.045     -5.097      0.000      -0.317      -0.140
==============================================================================
Omnibus:                        6.436   Durbin-Watson:                   0.845
Prob(Omnibus):                  0.040   Jarque-Bera (JB):                6.643
Skew:                           0.495   Prob(JB):                       0.0361
Kurtosis:                       2.844   Cond. No.                     4.78e+03
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.78e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

실습 

In [13]:
df4 = pd.read_excel('small_used_car.xlsx')

In [14]:
df4.head()

mileage model  price  year  my_car_damage  other_car_damage
0    63608    K3    970  2017              0            564596
1    69336    K3   1130  2015        1839700           1140150
2    36000    K3   1380  2016         446520           2244910
3    19029    K3   1390  2017         889000           4196110
4    97090    K3    760  2015        2339137           2029570

In [15]:
df4.model.groupby

<bound method Series.groupby of 0          K3
1          K3
2          K3
3          K3
4          K3
        ...  
269    Avante
270    Avante
271    Avante
272    Avante
273    Avante
Name: model, Length: 274, dtype: object>

In [16]:
ols('price ~ mileage + model + year + my_car_damage + other_car_damage', data = df4).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  price   R-squared:                       0.783
Model:                            OLS   Adj. R-squared:                  0.779
Method:                 Least Squares   F-statistic:                     193.2
Date:                Tue, 06 Oct 2020   Prob (F-statistic):           1.14e-86
Time:                        14:54:45   Log-Likelihood:                -1770.2
No. Observations:                 274   AIC:                             3552.
Df Residuals:                     268   BIC:                             3574.
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
====================================================================================
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
Intercept        -1.523e+05   9586.614    -15.885      0.000   -1.71e+05   -1.33e+05
model[T.K3]        -32.7958     22.205     -1.477      0.141     -76.514      10.923
mileage             -0.0024      0.000     -8.958      0.000      -0.003      -0.002
year                76.1874      4.755     16.022      0.000      66.825      85.550
my_car_damage    -1.619e-05   3.31e-06     -4.894      0.000   -2.27e-05   -9.68e-06
other_car_damage -6.073e-06   2.14e-06     -2.836      0.005   -1.03e-05   -1.86e-06
==============================================================================
Omnibus:                       11.358   Durbin-Watson:                   1.553
Prob(Omnibus):                  0.003   Jarque-Bera (JB):               14.249
Skew:                           0.347   Prob(JB):                     0.000805
Kurtosis:                       3.875   Cond. No.                     5.46e+09
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 5.46e+09. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [17]:
-1.523e+05 -0.0024 * 100000 + 76.1874*2012 -1.619e-05*200000 -6.073e-06*1000000

739.7377999999895

# 다중회귀분석 연습
- 안전도 데이터를 다운받아 risk를 종속변수로, length, horsepower, city_mpg, doors을 독립변수로 회귀분석을 해보세요.

In [18]:
df3.head()

risk fuel aspiration doors   body wheels engine_location  wheel_base  \
0     2  gas        std  four  sedan    fwd           front        99.8   
1     2  gas        std  four  sedan    4wd           front        99.4   
2     1  gas        std  four  sedan    fwd           front       105.8   
3     1  gas      turbo  four  sedan    fwd           front       105.8   
4     2  gas        std   two  sedan    rwd           front       101.2   

   length  width  height  curb_weight  bore  stroke  compression_ratio  \
0   176.6   66.2    54.3         2337  3.19     3.4               10.0   
1   176.6   66.4    54.3         2824  3.19     3.4                8.0   
2   192.7   71.4    55.7         2844  3.19     3.4                8.5   
3   192.7   71.4    55.9         3086  3.13     3.4                8.3   
4   176.8   64.8    54.3         2395  3.50     2.8                8.8   

   horsepower  peak_rpm  city_mpg  highway_mpg  
0         102      5500        24           30  
1         115      5500        18           22  
2         110      5500        19           25  
3         140      5500        17           20  
4         101      5800        23           29

In [19]:
ols('risk ~ length + horsepower + city_mpg + doors', data = df3).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   risk   R-squared:                       0.413
Model:                            OLS   Adj. R-squared:                  0.397
Method:                 Least Squares   F-statistic:                     27.03
Date:                Tue, 06 Oct 2020   Prob (F-statistic):           5.34e-17
Time:                        14:54:46   Log-Likelihood:                -210.89
No. Observations:                 159   AIC:                             431.8
Df Residuals:                     154   BIC:                             447.1
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
Intercept        4.8771      2.196      2.221      0.028       0.539       9.215
doors[T.two]     1.3104      0.177      7.401      0.000       0.961       1.660
length          -0.0261      0.011     -2.424      0.017      -0.047      -0.005
horsepower       0.0037      0.005      0.795      0.428      -0.006       0.013
city_mpg        -0.0200      0.024     -0.828      0.409      -0.068       0.028
==============================================================================
Omnibus:                        3.391   Durbin-Watson:                   0.867
Prob(Omnibus):                  0.183   Jarque-Bera (JB):                3.410
Skew:                           0.324   Prob(JB):                        0.182
Kurtosis:                       2.693   Cond. No.                     5.99e+03
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 5.99e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [20]:
4.8771 -0.0261*176.6 + 0.0037*102 -0.0200*24

0.1652400000000005

In [21]:
ols('risk ~ length + city_mpg + doors', data = df3).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   risk   R-squared:                       0.410
Model:                            OLS   Adj. R-squared:                  0.399
Method:                 Least Squares   F-statistic:                     35.92
Date:                Tue, 06 Oct 2020   Prob (F-statistic):           1.11e-17
Time:                        14:54:46   Log-Likelihood:                -211.22
No. Observations:                 159   AIC:                             430.4
Df Residuals:                     155   BIC:                             442.7
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
Intercept        5.1228      2.171      2.359      0.020       0.833       9.412
doors[T.two]     1.3542      0.168      8.057      0.000       1.022       1.686
length          -0.0236      0.010     -2.294      0.023      -0.044      -0.003
city_mpg        -0.0328      0.018     -1.833      0.069      -0.068       0.003
==============================================================================
Omnibus:                        3.993   Durbin-Watson:                   0.890
Prob(Omnibus):                  0.136   Jarque-Bera (JB):                3.968
Skew:                           0.347   Prob(JB):                        0.138
Kurtosis:                       2.659   Cond. No.                     5.18e+03
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 5.18e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [22]:
res = ols('risk ~ length + city_mpg + doors', data = df3).fit()

In [23]:
res.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   risk   R-squared:                       0.410
Model:                            OLS   Adj. R-squared:                  0.399
Method:                 Least Squares   F-statistic:                     35.92
Date:                Tue, 06 Oct 2020   Prob (F-statistic):           1.11e-17
Time:                        14:54:46   Log-Likelihood:                -211.22
No. Observations:                 159   AIC:                             430.4
Df Residuals:                     155   BIC:                             442.7
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
Intercept        5.1228      2.171      2.359      0.020       0.833       9.412
doors[T.two]     1.3542      0.168      8.057      0.000       1.022       1.686
length          -0.0236      0.010     -2.294      0.023      -0.044      -0.003
city_mpg        -0.0328      0.018     -1.833      0.069      -0.068       0.003
==============================================================================
Omnibus:                        3.993   Durbin-Watson:                   0.890
Prob(Omnibus):                  0.136   Jarque-Bera (JB):                3.968
Skew:                           0.347   Prob(JB):                        0.138
Kurtosis:                       2.659   Cond. No.                     5.18e+03
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 5.18e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [24]:
res.predict(df3)

0      0.175017
1      0.372104
2     -0.039941
3      0.025755
4      1.557335
         ...   
154   -0.079477
155    0.051914
156    0.084762
157   -0.178020
158    0.051914
Length: 159, dtype: float64

In [25]:
new_df = pd.read_excel('새차.xlsx')

In [26]:
res.predict(new_df)

0    0.175017
1    1.547662
dtype: float64

In [44]:
df = pd.read_excel('야매 데이터.xlsx')

In [45]:
df.head()

café  cake  coffee  power  toilet  study  accessibility  price  preference
0  스타벅스     5       5      5       5      5              5      3           5
1   이디야     3       2      3       3      3              5      4           4
2   할리스     3       3      4       3      5              4      3           3
3    탐탐     4       4      5       3      5              3      3           4
4   빽다방     2       3      1       5      1              4      5           1

In [46]:
ols('preference ~  cake + coffee + power + toilet + study + accessibility + price + preference', data = df).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:             preference   R-squared:                       1.000
Model:                            OLS   Adj. R-squared:                  1.000
Method:                 Least Squares   F-statistic:                 5.617e+28
Date:                Tue, 06 Oct 2020   Prob (F-statistic):          3.00e-156
Time:                        15:05:48   Log-Likelihood:                 623.75
No. Observations:                  20   AIC:                            -1230.
Df Residuals:                      11   BIC:                            -1221.
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept     -3.553e-15   1.93e-14     -0.184      0.857    -4.6e-14    3.89e-14
cake           -2.22e-16   4.23e-15     -0.053      0.959   -9.53e-15    9.09e-15
coffee        -2.914e-16   2.45e-15     -0.119      0.907   -5.68e-15     5.1e-15
power         -3.331e-16   1.91e-15     -0.175      0.864   -4.53e-15    3.86e-15
toilet         4.857e-16   2.21e-15      0.219      0.830   -4.39e-15    5.36e-15
study                  0   2.56e-15          0      1.000   -5.64e-15    5.64e-15
accessibility -8.882e-16   1.86e-15     -0.478      0.642   -4.98e-15     3.2e-15
price          4.441e-16   3.14e-15      0.141      0.890   -6.46e-15    7.35e-15
preference        1.0000   1.85e-15   5.42e+14      0.000       1.000       1.000
==============================================================================
Omnibus:                        0.084   Durbin-Watson:                   0.081
Prob(Omnibus):                  0.959   Jarque-Bera (JB):                0.177
Skew:                           0.124   Prob(JB):                        0.915
Kurtosis:                       2.612   Cond. No.                         88.4
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [50]:
res = ols('preference ~ cake + coffee + power + toilet + study + accessibility + price + preference', data = df).fit()

In [51]:
res.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:             preference   R-squared:                       1.000
Model:                            OLS   Adj. R-squared:                  1.000
Method:                 Least Squares   F-statistic:                 5.617e+28
Date:                Tue, 06 Oct 2020   Prob (F-statistic):          3.00e-156
Time:                        15:06:04   Log-Likelihood:                 623.75
No. Observations:                  20   AIC:                            -1230.
Df Residuals:                      11   BIC:                            -1221.
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept     -3.553e-15   1.93e-14     -0.184      0.857    -4.6e-14    3.89e-14
cake           -2.22e-16   4.23e-15     -0.053      0.959   -9.53e-15    9.09e-15
coffee        -2.914e-16   2.45e-15     -0.119      0.907   -5.68e-15     5.1e-15
power         -3.331e-16   1.91e-15     -0.175      0.864   -4.53e-15    3.86e-15
toilet         4.857e-16   2.21e-15      0.219      0.830   -4.39e-15    5.36e-15
study                  0   2.56e-15          0      1.000   -5.64e-15    5.64e-15
accessibility -8.882e-16   1.86e-15     -0.478      0.642   -4.98e-15     3.2e-15
price          4.441e-16   3.14e-15      0.141      0.890   -6.46e-15    7.35e-15
preference        1.0000   1.85e-15   5.42e+14      0.000       1.000       1.000
==============================================================================
Omnibus:                        0.084   Durbin-Watson:                   0.081
Prob(Omnibus):                  0.959   Jarque-Bera (JB):                0.177
Skew:                           0.124   Prob(JB):                        0.915
Kurtosis:                       2.612   Cond. No.                         88.4
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [49]:
res.predict(df)

0     5.0
1     4.0
2     3.0
3     4.0
4     1.0
5     3.0
6     5.0
7     3.0
8     4.0
9     5.0
10    3.0
11    2.0
12    1.0
13    5.0
14    3.0
15    4.0
16    5.0
17    5.0
18    5.0
19    1.0
dtype: float64

In [52]:
df4.head()

mileage model  price  year  my_car_damage  other_car_damage
0    63608    K3    970  2017              0            564596
1    69336    K3   1130  2015        1839700           1140150
2    36000    K3   1380  2016         446520           2244910
3    19029    K3   1390  2017         889000           4196110
4    97090    K3    760  2015        2339137           2029570

# 단계적 회귀분석 실습

## 전진선택 1단계

In [62]:
df4.head()

mileage model  price  year  my_car_damage  other_car_damage
0    63608    K3    970  2017              0            564596
1    69336    K3   1130  2015        1839700           1140150
2    36000    K3   1380  2016         446520           2244910
3    19029    K3   1390  2017         889000           4196110
4    97090    K3    760  2015        2339137           2029570

In [66]:
df4.columns

Index(['mileage', 'model', 'price', 'year', 'my_car_damage',
       'other_car_damage'],
      dtype='object')

In [69]:
res = ols('price ~ mileage', data = df4).fit()
res.aic

3795.387174945697

In [70]:
res = ols('price ~ model', data = df4).fit()
res.aic

3959.75750148095

In [71]:
res = ols('price ~ year', data = df4).fit()
res.aic

3644.458936334708

In [83]:
res = ols('price ~ my_car_damage', data = df4).fit()
res.aic

3909.9641234834567

In [95]:
res = ols('price ~ other_car_damage', data = df4).fit()
res.aic

3945.6976134241786

-  year

In [92]:
result = []
for x in ['mileage', 'model', 'year', 'my_car_damage','other_car_damage']:
    formula = f'print ~ {x}'
    res = ols(formula, data=df4).fit()
    result.append((res.aic, formula))

PatsyError: Number of rows mismatch between data argument and print (274 versus 1)
    print ~ mileage
    ^^^^^

## 전진선택 2단계

In [86]:
res = ols('price ~ year + mileage', data = df4).fit()
res.aic

3586.4864382402584

In [87]:
res = ols('price ~ year +model', data = df4).fit()
res.aic

3644.282065360474

In [88]:
res = ols('price ~ year +  my_car_damage', data = df4).fit()
res.aic

3622.0496210245065

In [89]:
res = ols('price ~ year +  other_car_damage', data = df4).fit()
res.aic

3641.049597123469

- mileage